core/flow/slot.go

```diff
@@ -25,20 +26,23 @@ func (s *FlowSlot) Check(ctx *base.EntryContext) *base.TokenResult {
 			continue
 		}
 		r := canPassCheck(tc, ctx.StatNode, ctx.Input.AcquireCount)
+		if r == nil {
+			// nil means pass
+			continue
+		}
 		if r.Status() == base.ResultStatusBlocked {
-			return r
+			result.DeepCopyFrom(r)
```

Comment:
Why a copy is needed here?

Reply:
It's not needed, maybe return r is enough